core/src/main/scala/cilib/Archive.scala

```diff
@@ -0,0 +1,195 @@
+package cilib
+
+import scalaz._
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.numeric.Positive
+
+sealed trait Bound
+final case class Bounded(limit: Int Refined Positive) extends Bound
+final case class Unbounded() extends Bound
+
+sealed abstract class Archive[A: Order, B] {
```

Comment:
The `Order` constraint here needs to be placed on the usage functions - keeping it on the class structure causes a lot of issues wrt usage.